### PR TITLE
More constructable items with wood:

### PR DESCRIPTION
**Pull request by ghost**
Wooden lattices (foundations) - thanks @nigglywiggly
Wooden platings (decks) - thanks @nigglywiggly
Wooden girders - thanks @nienhaus

### CODE REVIEW

**Comment by Kurfursten** [comment]
Being able to build a space station purely out of wood hurts muh immershuns

**Comment by ghost** [comment]
@kurfursten Can only use the wood lattices on snow and asteroids, won't work in space.

**Comment by Kurfursten** [comment]
Oh, okay

File: code/game/objects/items/stacks/sheets/sheet_types.dm [hunk 1]
```diff
@@ -204,6 +205,19 @@ var/global/list/datum/stack_recipe/wood_recipes = list (
 	sheettype = "wood"
 	w_type = RECYK_WOOD
 
+/obj/item/stack/sheet/wood/afterattack(atom/Target, mob/user, adjacent, params)
+	..()
+	if(adjacent)
+		if(isturf(Target) || istype(Target, /obj/structure/lattice))
```
Comment (Kurfursten) [inline code/game/objects/items/stacks/sheets/sheet_types.dm:211]: Where's the no-space check Potatoman?

File: code/game/objects/items/stacks/tiles/tile_types.dm [hunk 2]
```diff
@@ -70,6 +70,37 @@
 
 	material = "wood"
 
+/obj/item/stack/tile/wood/proc/build(turf/S as turf)
+	if(istype(S,/turf/space) || istype(S,/turf/unsimulated))
```
Comment (Kurfursten) [inline code/game/objects/items/stacks/tiles/tile_types.dm]: You said there wouldn't be space!

**Comment by ihadtoregisterforthis** [comment]
smaller wooden station

File: code/game/turfs/space/space.dm [hunk 3]
```diff
@@ -25,8 +25,8 @@
 	return locate(/obj/structure/lattice) in contents
 
 
-/turf/space/canBuildLattice()
-	if(!(locate(/obj/structure/lattice) in contents))
+/turf/space/canBuildLattice(var/material)
+	if(!(locate(/obj/structure/lattice) in contents) && !(istype(material,/obj/item/stack/sheet/wood)))
```
Comment (ghost) [inline code/game/turfs/space/space.dm:29]: Right here, @Kurfursten.

**Comment by FudgePucker** [comment]
> tfw can't make autism burnchamber in space made entirely out of wood

*buzz

File: code/game/objects/items/stacks/tiles/tile_types.dm (hunk 2 above)
Comment (ghost) [inline code/game/objects/items/stacks/tiles/tile_types.dm]: I-i copy pasted this from the other tile.

**Comment by ConkerDev** [comment]
Make lattices constuctable with wood
Make wood tiles be able to be placed over lattices
Just make them flammable or something

File: code/game/objects/items/stacks/sheets/sheet_types.dm [hunk 4]
```diff
@@ -204,6 +205,19 @@ var/global/list/datum/stack_recipe/wood_recipes = list (
...
+			var/turf/T = get_turf(Target)
+			var/obj/item/stack/sheet/wood = src
```
Comment (clusterfack) [inline code/game/objects/items/stacks/sheets/sheet_types.dm]: You don't need to typecast src just use the procs normally

File: code/game/objects/items/stacks/tiles/tile_types.dm [hunk 5]
```diff
@@ -70,6 +70,37 @@
 
 	material = "wood"
 
+/obj/item/stack/tile/wood/proc/build(turf/S as turf)
+	if(istype(S,/turf/unsimulated))
+		S.ChangeTurf(/turf/simulated/floor/plating/deck/airless)
+	else
+		S.ChangeTurf(/turf/simulated/floor/plating/deck)
+	return
```
Comment (clusterfack) [inline code/game/objects/items/stacks/tiles/tile_types.dm]: This is a trailing return

File: code/game/objects/items/stacks/tiles/tile_types.dm [hunk 6]
```diff
@@ -70,6 +70,37 @@
 
 	material = "wood"
 
+/obj/item/stack/tile/wood/proc/build(turf/S as turf)
+	if(istype(S,/turf/unsimulated))
+		S.ChangeTurf(/turf/simulated/floor/plating/deck/airless)
```
Comment (clusterfack) [inline code/game/objects/items/stacks/tiles/tile_types.dm]: Why are you making this an airless simulated turf, sure it makes sense for asteroids maybe but lots of unsim turfs have air

File: code/game/objects/structures/girders.dm [hunk 7]
```diff
@@ -15,9 +28,9 @@
 				if(do_after(user, src, 40))
 					user.visible_message("<span class='warning'>[user] dissasembles \the [src]</span>", \
 					"<span class='notice'>You dissasemble \the [src]</span>")
-					getFromPool(/obj/item/stack/sheet/metal, get_turf(src))
+					var/stack = (state==3) ?  "/obj/item/stack/sheet/wood" : "/obj/item/stack/sheet/metal"
```
Comment (clusterfack) [inline code/game/objects/structures/girders.dm]: Jesus use an actual var to hold what material its made out of

File: code/game/objects/structures/girders.dm [hunk 8]
```diff
@@ -115,7 +129,7 @@
 			state = 1
 			update_icon()
 
-	else if(istype(W, /obj/item/weapon/crowbar) && state == 0 && anchored) //Turning normal girder into disloged girder
+	else if(istype(W, /obj/item/weapon/crowbar) && (state == 0 || state == 3) && anchored) //Turning normal girder into disloged girder
```
Comment (clusterfack) [inline code/game/objects/structures/girders.dm]: if you want to prevent this make a can_reinforce var or some shit

**Comment by clusterfack** [comment]
Where are the images of all these things celt

File: code/game/objects/items/stacks/tiles/tile_types.dm (hunk 6 above)
Comment (ghost) [inline code/game/objects/items/stacks/tiles/tile_types.dm:75]: wouldn't it be better they fill with the air of the unsim turf rather than their own?

**Comment by ghost** [comment]
AAAAAAAAAA

**Comment by ghost** [comment]
wood

File: code/game/objects/items/stacks/tiles/tile_types.dm (hunk 6 above)
Comment (clusterfack) [inline code/game/objects/items/stacks/tiles/tile_types.dm]: Okay, so where is the code to do that

**Comment by ghost** [comment]
N3X15: Needs a :+1: celt

:+1:

File: code/game/objects/structures/lattice.dm [hunk 9]
```diff
@@ -43,9 +43,18 @@
 	if(iswelder(C))
 		var/obj/item/weapon/weldingtool/WeldingTool = C
 		if(WeldingTool.remove_fuel(0, user))
-			to_chat(user, "<span class='notice'>Slicing lattice joints...</span>")
+			to_chat(user, "<span class='notice'>Slicing [src] joints...</span>")
 		new/obj/item/stack/rods(loc)
```
Comment (Intigracy) [inline code/game/objects/structures/lattice.dm:47]: Isn't this going to give you rods even if it's made of wood?

Comment (clusterfack) [inline code/game/objects/structures/lattice.dm]: Also *using a welder to chop up wood*

**Comment by ghost** [comment]
there! I fixed it!

File: code/game/objects/structures/lattice.dm [hunk 10]
```diff
@@ -43,9 +43,27 @@
...
 		qdel(src)
 	else
 		var/turf/T = get_turf(src)
 		T.attackby(C, user) //Attacking to the lattice will attack to the space turf
+
+/obj/structure/lattice/wood/attackby(obj/item/C as obj, mob/user as mob)
+	if((C.is_sharp() >= 1.2) && (C.w_class >= 2) && do_after(user, src, 10)) // If C is able to cut down a tree and this nerd actually cuts it down.
+		new/obj/item/stack/sheet/wood(loc)
+		to_chat(user, "<span class='notice'>You chop the [src] apart!</span>")
```
Comment (ghost) [inline code/game/objects/structures/lattice.dm]: All fixed!

**Comment by Intigracy** [comment]
Now add a changelog.

**Comment by ghost** [comment]
I don't want to pollute the changelog with silly changes

**Comment by Intigracy** [comment]
It's adding things you can build.

Changelog it you fag.

**Comment by Intigracy** [comment]
Also why would I build a wood girder when I can just make a wood wall on top of a metal girder?

**Comment by ghost** [comment]
When you have more wood than metal.

**Comment by ghost** [comment]
[spoiler] on a snow planet [/spoiler]

**Comment by Zth--** [comment]
:+1:

**Pull request merged**